src/lib/protocols/kerberos.c

```diff
@@ -200,17 +200,21 @@ static int krb_decode_asn1_blocks_skip(struct ndpi_detection_module_struct *ndpi
   return length;
 }
 
-static void strncpy_lower(char * const dst, size_t dst_siz,
-                          char const * const src, size_t src_siz)
+static void krb_strncpy_lower(char * const dst, size_t dst_siz,
+                              char const * const src, size_t src_siz)
 {
    int dst_len = ndpi_min(src_siz, dst_siz - 1);
 
-   strncpy(dst, src, dst_len);
    dst[dst_len] = '\0';
 
    for(int i = 0; i < dst_len; ++i)
    {
-     dst[i] = tolower(dst[i]);
+     if (ndpi_isprint(src[i]) == 0)
```

Comment:
Not sure how we should deal with non printable characters in a printable string, but printing binary data in a text terminal is in general not a good idea can lead to different results on different platforms. We could implement an "escape" function that prints something e.g. "\\x90" if a buffer contains 0x90.